### PR TITLE
refactor and cleanup

### PR DESCRIPTION
**Pull request by duanyutong**
### Background

refactor and cleanup

### What's new
- ✅ refactor and cleanup
- ✅ publish franka exception details over lcm before and regardless of `RecoverFromControlException` outcome
- ✅ resolve a ton of linter warnings, including the include style
- ✅ updated hooks including license header
- ❌ New feature is accompanied by new test: [name and description of new test].

### Related work
- ❌ This PR needs [link]
- ❌ [link] needs this PR

### TODOs / Nice-To-Haves
- ❌ [Add new unit test for new feature.]
- ❌ [Add new system test for new feature.]

### Tests
1. ✅ Tested on simulated robot: [Describe the tests/commands used.]
2. ✅ Tested on physical robot: [Describe the tests/commands used.]

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅ New code passed `clang-format` and `cpplint` at least, if not all of our static code analysers.
6. ✅ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: src/driver/franka_plan_runner.cc
Comment (duanyutong) [inline src/driver/franka_plan_runner.cc:216]: main loop refactor starts here

File: src/driver/franka_plan_runner.cc
Comment (duanyutong) [inline src/driver/franka_plan_runner.cc:308]: publish franka exception details over lcm before and regardless of `RecoverFromControlException` outcome

File: src/utils/util_conv.h
Comment (dmsj) [inline src/utils/util_conv.h:43]: remove note

File: src/driver/franka_plan_runner.cc
Comment (dmsj) [inline src/driver/franka_plan_runner.cc:324]: How should we specify these wait times? We have a 1-second wait earlier and a 10-second wait here.

**Review by dmsj — COMMENTED** [review]
looks pretty good! needs on-robot testing for full sign-off

File: src/driver/communication_interface.cc
Comment (5yler) [inline src/driver/communication_interface.cc:45]: "opause status"?

File: src/driver/communication_interface.h
Comment (5yler) [inline src/driver/communication_interface.h:113]: why?

File: src/driver/franka_plan_runner.cc
Comment (5yler) [reply src/driver/franka_plan_runner.cc:228]: If there's a networking problem, there's no point in retrying more frequently. If we hit this code block something is very wrong and the controller is likely not connected or not communicating

File: src/driver/franka_plan_runner.cc
Comment (5yler) [reply src/driver/franka_plan_runner.cc:308]: The driver status LCM message with `success = false` indicates that the driver is not running. A successful recovery from control exception shouldn't have a success of `false`

**Review by 5yler — COMMENTED** [review]
:heart_eyes_cat:

File: src/driver/franka_plan_runner.cc
Comment (duanyutong) [reply src/driver/franka_plan_runner.cc]: updated.

File: src/utils/util_conv.h
Comment (duanyutong) [reply src/utils/util_conv.h:43]: removed

File: src/driver/franka_plan_runner.cc
Comment (duanyutong) [reply src/driver/franka_plan_runner.cc:271]: they are different things, this is only for the status printout

File: src/driver/communication_interface.h
Comment (duanyutong) [reply src/driver/communication_interface.h:113]: typo? changed to pause

File: src/driver/communication_interface.cc
Comment (duanyutong) [reply src/driver/communication_interface.cc:45]: typo? changed to pause

**Review by 5yler — APPROVED** [review]
LGTM platform & style assuming sufficiently tested on robot

**Pull request merged**
